core/src/main/scala/org/apache/spark/storage/BlockInfoManager.scala

```diff
@@ -502,7 +536,10 @@ private[storage] class BlockInfoManager extends Logging {
         throw new IllegalStateException(
           s"Task $taskAttemptId called remove() on block $blockId without a write lock")
       } else {
-        blockInfoWrappers.remove(blockId)
+        invisibleRDDBlocks.synchronized {
+          blockInfoWrappers.remove(blockId)
```

Comment:
`blockInfoWrappers` isn't necessary to be protected by `invisibleRDDBlocks.synchronized`?

Reply:
Just want to make sure that state change for both `blockInfoWrappers` and `invisibleRDDBlocks` are processed in the same syncrhonized block to avoid potential concurrent issues since we depends on the state of the 2 variables to decide whether a block is visible.